src/HideLooseAssetService.cpp

```diff
@@ -19,13 +23,35 @@ namespace BsaPacker
 			return false;
 		}
 
+		// Only hide loose files that were not excluded when creating the archive (blacklisted), so might still be needed by the mod
+		QStringList blacklistExtensions = this->m_SettingsService->GetPluginSetting(SettingsService::SETTING_BLACKLISTED_FILES).toString().split(';');
+		for (auto& ext : blacklistExtensions) {
+			ext.prepend("*");
+		}
+
+		// Hide subdirectories
 		const QString& absModDir = modDirectory.absolutePath();
 		for (const QString& subDir : modDirectory.entryList(QDir::Dirs | QDir::NoDotAndDotDot)) {
 			const QString& absPath = absModDir + '/' + subDir;
 			QDir dir(absPath);
-			if (!dir.dirName().endsWith(".mohidden")) {
-				dir.rename(absPath, absPath + ".mohidden");
+			if (dir.dirName().endsWith(s_HiddenExt) || dir.isEmpty()) {
+				continue;
+			}
+			if (!dir.rename(absPath, absPath + s_HiddenExt)) {
+				qWarning() << "Failed to hide directory " << absPath;
+			}
+		}
+
+		// Restore files with blacklisted extension to their original directories
+		QDir renamerDir(absModDir);
+		QDirIterator iterator(absModDir, blacklistExtensions, QDir::Files, QDirIterator::Subdirectories);
+		while (iterator.hasNext()) {
+			const QFileInfo& fileInfo(iterator.nextFileInfo());
+			if (!renamerDir.mkpath(fileInfo.absolutePath().replace(s_HiddenExt, ""))) {
+				qWarning() << "Failed to make directory " << fileInfo.absolutePath().replace(s_HiddenExt, "");
+				continue;
 			}
+			renamerDir.rename(fileInfo.absoluteFilePath(), fileInfo.absoluteFilePath().replace(s_HiddenExt, ""));
```

Comment:
These `replace` are stripping `.mohidden` from the entire path, so potentially even from later down the tree, not just the top level.

If you have a blacklisted file that ends with `.mohidden`, then you are not only restoring it to the original unhidden folder but also removing the `.mohidden` part, which is likely to be unwanted behaviour. 

You should only strip `.mohidden` from the first subdir name.

To make this simpler I think you could iterate over the hidden subdirs and find all blacklisted children inside each hidden subdir, and rename those, this way you have already the paths split at the right places:
- `parentDir / hiddenDir / blacklistedFilePath`
Which you change to :
- `parentDir / unhiddenDir / blacklistedFilePath`

Should also avoid iterating over the now unhidden blacklisted files that you already renamed.

Reply:
Good points. I have moved the iterator to be per hidden subdirectory. I also adjusted the iterator loop to address your comments. Let me know if you see any other issues (or new ones).

I also removed the gameDependency requirement for now. I overlooked some games that should be supported. I need to do more research about the games MO2 and the BSA Packer support before readding it. Using the tool on an unsupported game is handled gracefully, so it should be fine being visible in those instances.